src/Core/Geographic/Coordinates.js

```diff
@@ -1,441 +1,270 @@
- * // returns 2.33
- *
- * // or
- *
- * const position = { x: 20885167, y: 849862, z: 23385912 };
- * const coords = new Coordinates('EPSG:4978', position.x, position.y, position.z);  // Geocentric system
- * const coordinates = coords.as('EPSG:4326');  // Geographic system
- * coordinates.longitude(); // Longitude in geographic system
- * // returns 2.330201911389028
- *
- * @return     {number} - The longitude of the position.
- */
-
-Coordinates.prototype.longitude = function longitude() {
-    _assertIsGeographic(this.crs);
-    return this._values[0];
-};
-
-/**
- * Returns the latitude in geographic coordinates. Coordinates must be in geographic system (can be converted by using {@linkcode as()} ).
- * @example
- *
- * const position = { longitude: 2.33, latitude: 48.24, altitude: 24999549 };
- * const coordinates = new Coordinates('EPSG:4326', position.longitude, position.latitude, position.altitude); // Geographic system
- * coordinates.latitude(); // Latitude in geographic system
- * // returns : 48.24
- *
- * // or
- *
- * const position = { x: 20885167, y: 849862, z: 23385912 };
- * const coords = new Coordinates('EPSG:4978', position.x, position.y, position.z);  // Geocentric system
- * const coordinates = coords.as('EPSG:4326');  // Geographic system
- * coordinates.latitude(); // Latitude in geographic system
- * // returns : 48.24830764643365
- *
- * @return     {number} - The latitude of the position.
- */
-
-Coordinates.prototype.latitude = function latitude() {
-    return this._values[1];
-};
-
-/**
- * Returns the altitude in geographic coordinates. Coordinates must be in geographic system (can be converted by using {@linkcode as()} ).
- * @example
- *
- * const position = { longitude: 2.33, latitude: 48.24, altitude: 24999549 };
- * const coordinates = new Coordinates('EPSG:4326', position.longitude, position.latitude, position.altitude); // Geographic system
- * coordinates.altitude(); // Altitude in geographic system
- * // returns : 24999549
- *
- * // or
- *
- * const position = { x: 20885167, y: 849862, z: 23385912 };
- * const coords = new Coordinates('EPSG:4978', position.x, position.y, position.z);  // Geocentric system
- * const coordinates = coords.as('EPSG:4326');  // Geographic system
- * coordinates.altitude(); // Altitude in geographic system
- * // returns : 24999548.046711832
- *
- * @return     {number} - The altitude of the position.
- */
 
-Coordinates.prototype.altitude = function altitude() {
-    _assertIsGeographic(this.crs);
-    return this._values[2];
-};
+    /**
+     * Returns a new Coordinates with the same values as this one. A target can
+     * be passed to avoid the creation of a new one. Otherwise it will
+     * instantiate a new Coordinates with the same CRS as this one.
+     *
+     * @param {Coordinates} [target]
+     *
+     * @return {Coordinates} The target with its new coordinates.
+     */
+    clone(target = new Coordinates(this.crs)) {
```

Comment:
Target isn't necessary because copy do the same thing.

Reply:
Yeah I agree, but I think it doesn't hurt to leave both of them, no ?

Reply:
the function must itself instantiate `Coordinates` and not have target.

Reply:
you could write  
```
return target.set(this.x, this.y, this.z);
```
there are other functions where is possible to simplify as well.